module/zfs/sa.c

```diff
@@ -451,6 +451,54 @@ sa_add_layout_entry(objset_t *os, sa_attr_type_t *attrs, int attr_count,
 	return (tb);
 }
 
+static sa_lot_t *
+sa_lookup_cached_layout(sa_os_t *sa, uint64_t hash, sa_attr_type_t *attrs,
+    int count)
+{
+	sa_layout_cache_t *slc;
+	sa_lot_t *tb = NULL;
+	int i, j;
+
+	kpreempt_disable();
```

Comment:
It's recommended in the Linux kernel to use `____cacheline_aligned` to get the right alignment, this way the right thing happens for each architecture.  I'm not sure if the other platforms definite a similar macro.